### PR TITLE
Nested Schema Categories for new Schema State

### PR DESCRIPTION
**Pull request by bacon1986**
This now works with a lot less code and without the emit.
image

Filtering updates the schema counts as expected:
image

~~It still suffers from the long schema name issue but I'll look into that now.~~ Fixed
image

Thanks

### CODE REVIEW

File: frontend/app/shared/components/schema-category.component.ts [hunk 1]
```diff
@@ -49,13 +51,19 @@ export class SchemaCategoryComponent implements OnChanges {
...
     public ngOnChanges() {
+        this.visibleCount = this.getCount(this.schemaCategory);
         if (this.schemaCategory.schemas.length < this.schemaCategory.schemaTotalCount) {
             this.isCollapsed = false;
         } else {
             this.isCollapsed = this.localStore.getBoolean(this.configKey());
         }
     }
 
+    private getCount(category: SchemaCategory): number {
```
Comment (bacon1986) [inline frontend/app/shared/components/schema-category.component.ts:62]: This method replaced the previous `emit` which allowed the controls to update either other when they filtered, but now the filtering is all in the state that's not needed any more 👍

**Comment by bacon1986** [comment]
I've just noticed that adding multi-level categories at the bottom of the schema list doesn't work so I'll look into that.

**Comment by pjmagee** [comment]
Okay, this is really nice actually.

**Comment by SebastianStehle** [comment]
I have merged it and made a few improvements to also calculate the filter count outside of the components.

**Pull request merged**